diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2da792d53..2f2d076c6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1118,17 +1118,10 @@ configure_file(
 
 # Installation Directories
 # ========================
-if(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
-  set(FLB_INSTALL_BINDIR "bin")
-  set(FLB_INSTALL_LIBDIR "lib")
-  set(FLB_INSTALL_CONFDIR "conf")
-  set(FLB_INSTALL_INCLUDEDIR "include")
-else()
-  set(FLB_INSTALL_BINDIR ${CMAKE_INSTALL_FULL_BINDIR})
-  set(FLB_INSTALL_LIBDIR "${CMAKE_INSTALL_LIBDIR}/${FLB_OUT_NAME}")
-  set(FLB_INSTALL_CONFDIR "${CMAKE_INSTALL_SYSCONFDIR}/${FLB_OUT_NAME}/")
-  set(FLB_INSTALL_INCLUDEDIR "${CMAKE_INSTALL_PREFIX}/include")
-endif()
+set(FLB_INSTALL_BINDIR "${CMAKE_INSTALL_BINDIR}")
+set(FLB_INSTALL_LIBDIR "${CMAKE_INSTALL_LIBDIR}")
+set(FLB_INSTALL_CONFDIR "${CMAKE_INSTALL_SYSCONFDIR}/${FLB_OUT_NAME}/")
+set(FLB_INSTALL_INCLUDEDIR "${CMAKE_INSTALL_PREFIX}/include")
 
 # Instruct CMake to build the Fluent Bit Core
 add_subdirectory(include)
