diff --git a/Makefile b/Makefile
index ef24e94..6f3ddb6 100644
--- a/Makefile
+++ b/Makefile
@@ -136,7 +136,7 @@ libxxhash: $(LIBXXH)
 
 .PHONY: lib
 lib:  ## generate static and dynamic xxhash libraries
-lib: libxxhash.a libxxhash
+lib: libxxhash.a
 
 # helper targets
 
@@ -446,9 +446,6 @@ install: lib libxxhash.pc xxhsum  ## install libraries, CLI, links and man page
 	@echo Installing libxxhash
 	$(Q)$(INSTALL) -d -m 755 $(DESTDIR)$(LIBDIR)
 	$(Q)$(INSTALL_DATA) libxxhash.a $(DESTDIR)$(LIBDIR)
-	$(Q)$(INSTALL_PROGRAM) $(LIBXXH) $(DESTDIR)$(LIBDIR)
-	$(Q)ln -sf $(LIBXXH) $(DESTDIR)$(LIBDIR)/libxxhash.$(SHARED_EXT_MAJOR)
-	$(Q)ln -sf $(LIBXXH) $(DESTDIR)$(LIBDIR)/libxxhash.$(SHARED_EXT)
 	$(Q)$(INSTALL) -d -m 755 $(DESTDIR)$(INCLUDEDIR)   # includes
 	$(Q)$(INSTALL_DATA) xxhash.h $(DESTDIR)$(INCLUDEDIR)
 	$(Q)$(INSTALL_DATA) xxh3.h $(DESTDIR)$(INCLUDEDIR) # for compatibility, will be removed in v0.9.0
