diff --git a/dev/restinio/expected.hpp b/dev/restinio/expected.hpp
--- a/dev/restinio/expected.hpp
+++ b/dev/restinio/expected.hpp
@@ -10,14 +10,16 @@

 #pragma once

-#include <nonstd/expected.hpp>
+#include <expected>

 namespace restinio
 {
-	template< typename T, typename E >
-	using expected_t = nonstd::expected<T, E>;
+template <typename T, typename E>
+using expected_t = std::expected<T, E>;

-	using nonstd::make_unexpected;
+template <typename U>
+auto make_unexpected(U&& u) {
+  return std::unexpected(std::forward<U>(u));
+}

 } /* namespace restinio */
-
